library/core/src/ptr/non_null.rs

```diff
@@ -1829,4 +1854,59 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+  
+    #[kani::proof_for_contract(NonNull::byte_sub)]
+    pub fn non_null_check_byte_sub() {
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator = PointerGenerator::<SIZE>::new();
+        let count: usize = kani::any();
+        let raw_ptr: *mut i32 = generator.any_in_bounds().ptr as *mut i32;
+
+        unsafe {
+            let ptr = NonNull::new(raw_ptr).unwrap();
+            let result = ptr.byte_sub(count);
+        }
+    }
+
+  #[kani::proof_for_contract(NonNull::offset)]
+    pub fn non_null_check_offset(){
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator1 = PointerGenerator::<SIZE>::new();
+        let mut generator2 = PointerGenerator::<SIZE>::new();
+        let start_ptr: *mut i32 = if kani::any() {
+            generator1.any_in_bounds().ptr as *mut i32
+        } else {
+            generator2.any_in_bounds().ptr as *mut i32
+        };
+        let ptr_nonnull = unsafe { NonNull::new(start_ptr).unwrap() };
+        let count: isize = kani::any();
+        unsafe { let result = ptr_nonnull.offset(count);
+       }
+}
+       
+    #[kani::proof_for_contract(NonNull::map_addr)]
+        pub fn non_null_check_map_addr() {
+        const SIZE: usize = mem::size_of::<i32>() * 100;
+        let mut generator = PointerGenerator::<SIZE>::new();
+        let ptr = generator.any_in_bounds().ptr as *mut i32;
+        let ptr_nonnull = unsafe { NonNull::new(ptr).unwrap() };
+        let new_offset: usize = kani::any_where(|&x| x < SIZE);
+        let f = |addr: NonZeroUsize| -> NonZeroUsize {
+        unsafe {NonZeroUsize::new_unchecked(addr.get().wrapping_add(new_offset)) 
```

Comment:
See my comment for the `with_addr` harness-- we don't know that the result of this pointer arithmetic will be nonzero, so calling `new_unchecked` on the result could trigger UB.
I would also recommend making this closure safe.